pinot-core/src/main/java/com/linkedin/pinot/core/operator/aggregation/function/AggregationFunction.java

```diff
@@ -17,87 +17,81 @@
 
 import com.linkedin.pinot.core.operator.aggregation.AggregationResultHolder;
 import com.linkedin.pinot.core.operator.aggregation.groupby.GroupByResultHolder;
-import java.util.List;
+import com.linkedin.pinot.core.operator.docvalsets.ProjectionBlockValSet;
+import java.io.Serializable;
+import javax.annotation.Nonnull;
 
 
 /**
- * Interface for Aggregation functions.
+ * Interface for aggregation functions.
  */
-public interface AggregationFunction {
+// TODO: revisit whether Serializable is necessary.
+public interface AggregationFunction<AggregationResult extends Serializable, FinalResult> {
```

Comment:
It is unclear what AggregationResult and FinalResult represent. Can you put some comments explaining the two. And  if required, rename AggregationResult to IntermediateResult.